types/from2/index.d.ts

```diff
@@ -8,21 +8,25 @@ import * as stream from 'stream';
 
 export = from2;
 
-declare function from2(read: from2.ReadInput): NodeJS.ReadableStream;
-declare function from2(opts: from2.ObjectModeOptions, read: from2.ReadObjectInput): NodeJS.ReadableStream;
-declare function from2(opts: from2.Options, read: from2.ReadInput): NodeJS.ReadableStream;
+declare function from2(read: from2.ReadInput): from2.Stream;
+declare function from2(opts: from2.ObjectModeOptions, read: from2.ReadObjectInput): from2.Stream;
+declare function from2(opts: from2.Options, read: from2.ReadInput): from2.Stream;
 
 declare namespace from2 {
-    function obj(read: ReadObjectInput): NodeJS.ReadableStream;
-    function obj(opts: { objectMode?: true | undefined } & stream.ReadableOptions, read: ReadObjectInput): NodeJS.ReadableStream;
+    interface Stream extends NodeJS.ReadableStream {
+        destroyed: boolean;
```

Comment:
I think it'd make sense to declare this as `readonly`:
```suggestion
        readonly destroyed: boolean;
```